File: GithubJavaPop/app/src/main/java/br/com/githubjavapop/adapter/PullRequestAdapter.java
package br.com.githubjavapop.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.squareup.picasso.Picasso;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.List;

import br.com.githubjavapop.R;
import br.com.githubjavapop.entidade.PullRequest;
import br.com.githubjavapop.viewHolder.PullRequestViewHolder;

public class PullRequestAdapter extends RecyclerView.Adapter<PullRequestViewHolder> implements
        View.OnClickListener {

    private List<PullRequest> listaPullRequest;
    private View.OnClickListener clickListener;

    public PullRequestAdapter(List<PullRequest> listaPullRequest) {
        this.listaPullRequest = listaPullRequest;
    }

    @NonNull
    @Override
    public PullRequestViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {

        LayoutInflater layoutInflater = LayoutInflater.from(viewGroup.getContext());
        View view = layoutInflater.inflate(R.layout.layout_pullrequest, viewGroup, false);
        view.setOnClickListener(this);

        return new PullRequestViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull PullRequestViewHolder pullRequestViewHolder, int i) {

        DateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
        String pullRequestData;

        if(listaPullRequest != null && listaPullRequest.size() > 0) {
            PullRequest pullRequest = listaPullRequest.get(i);
            pullRequestViewHolder.titulo.setText(pullRequest.getTitulo());
            pullRequestData = "Data de criação: " + dateFormat.format(pullRequest.getData());
            pullRequestViewHolder.data.setText(pullRequestData);
            pullRequestViewHolder.body.setText(pullRequest.getBody());
            pullRequestViewHolder.username.setText(pullRequest.getUsuario().getUsername());
            Picasso.get().load(pullRequest.getUsuario().getFoto()).into(pullRequestViewHolder.foto);
        }

    }

    @Override
    public int getItemCount() {
        return listaPullRequest.size();
    }

    public void setOnClickListener(View.OnClickListener clickListener) {
        this.clickListener = clickListener;
    }

    @Override
    public void onClick(View v) {
        if(clickListener != null) clickListener.onClick(v);
    }
}

File: app/src/main/java/matheusuehara/github/features/pullrequests/PullRequestAdapter.kt
package matheusuehara.github.features.pullrequests

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.lv_item_pull_request.view.*
import matheusuehara.github.R
import matheusuehara.github.data.model.PullRequest
import java.util.*

class PullRequestAdapter(var pullRequests: ArrayList<PullRequest>, var pullRequestClickListener: PullRequestClickListener) : RecyclerView.Adapter<PullRequestAdapter.ViewHolder>() {

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view), View.OnClickListener {
        var mPullRequestTitle: TextView = view.pull_request_title
        var mPullRequestBody: TextView = view.pull_request_body
        var mPullRequestUserName: TextView = view.pull_request_username
        var mPullRequestDate: TextView = view.pull_request_date
        var mPullRequestUseImage: ImageView = view.pull_request_user_image

        init {
            view.setOnClickListener(this)
        }

        override fun onClick(v: View) {
            pullRequestClickListener.onClick(pullRequests[adapterPosition])
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.lv_item_pull_request, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val pullRequest = pullRequests[position]
        holder.mPullRequestTitle.text = "[${pullRequest.state.toUpperCase()}] ${pullRequest.title}"
        holder.mPullRequestBody.text = pullRequest.body
        holder.mPullRequestDate.text = pullRequest.created_at
        holder.mPullRequestUserName.text = pullRequest.user.login
        Picasso.get().load(pullRequest.user.avatar_url).error(R.mipmap.ic_launcher).into(holder.mPullRequestUseImage)
    }

    override fun getItemCount(): Int {
        return pullRequests.size
    }

    fun addPullRequests( newPullRequests: ArrayList<PullRequest>){
        pullRequests.addAll(newPullRequests)
        pullRequests.sortByDescending { it.state}
        this.notifyDataSetChanged()
    }
}
